packages/cli/src/lib/server.ts

```diff
@@ -71,6 +75,39 @@ server.on('error', (err: Error) => {
 
 loadDestination(targetDirectory)
   .then((def) => {
+    const destination = new Destination(def as CloudDestinationDefinition)
+    const supportsDelete = destination.onDelete
+
+    if (supportsDelete) {
+      app.post(
+        '/delete',
+        asyncHandler(async (req: express.Request, res: express.Response) => {
+          spinner.start(chalk`Handling ${process.env.DESTINATION}#delete request`)
+
+          try {
+            if (destination.onDelete) {
+              await destination.onDelete(req.body.payload ?? {}, req.body.settings ?? {})
+            }
+
+            const debug = await getExchanges(destination.responses)
+            spinner.succeed(chalk`${destination.name} delete completed`)
+            return res.status(200).json(debug)
+          } catch (err) {
+            spinner.fail()
+            const error = err as ResponseError
+            let statusCode = error?.status ?? 500
+            let msg = error?.message
+
+            if (err instanceof HTTPError) {
+              statusCode = err?.response?.status ?? statusCode
```

Comment:
If it's an HTTPError, isn't it guaranteed to exist?
```suggestion
              statusCode = err.response?.status ?? statusCode
```